epuck_ros2_cpp/src/mpu9250.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright 1996-2020 Cyberbotics Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "epuck_ros2_cpp/mpu9250.hpp"
+
+#include <cmath>
+
+#define MPU9250_ADDRESS_AD1_0 0x68
+#define MPU9250_ADDRESS_AD1_1 0x69
+
+#define ACCEL_XOUT_H 0x3B
+#define GYRO_XOUT_H 0x43
+
+#define RAW2G (32768.0 / 2)
+#define G2MS2 (9.81)
+#define RAW2DEG (32768.0 / 250.0)
+
+MPU9250::MPU9250(std::shared_ptr<I2CWrapper> i2c) {
+  mI2c = i2c;
+  mAddress = MPU9250_ADDRESS_AD1_0;
+}
+
+int MPU9250::read_register(char reg, char *data, int size) {
+  if (mI2c->writeData(&reg, 1) != 1)
+    return -1;
+
+  if (mI2c->readData(data, size) != size)
+    return -2;
+
+  return 0;
+}
+
+void MPU9250::read_raw(int16_t *rawAccelerometer, int16_t *rawGyroscope) {
+  int status;
+  static char buffer[6];
+
+  for (int i = 0; i < 5; i++) {
+    status = mI2c->setAddress(mAddress);
+
+    status ^= read_register(ACCEL_XOUT_H, buffer, 6);
+    for (int i = 0; i < 3; i++)
```

Comment:
I just saw that this might be dangerous to use the same variable name (`i`) for the outer and inner loop, it would be safer to rename one of them.

Note also that declaring `status` outside the loop but using it only inside the loop can raise some warnings with some code checker (maybe cppcheck), but I assume you are doing this for performance reason and that's fine for me.

Reply:
Good catch, thanks!